module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -1167,20 +1167,51 @@ public function getMyHolds($patron)
             '/request-storage/requests',
             $query
         ) as $hold) {
-            $requestDate = date_create($hold->requestDate);
+            $requestDate = $this->dateConverter->convertToDisplayDate(
+                "Y-m-d H:i",
+                $hold->requestDate
+            );
             // Set expire date if it was included in the response
             $expireDate = isset($hold->requestExpirationDate)
-                ? date_create($hold->requestExpirationDate) : null;
+                ? $this->dateConverter->convertToDisplayDate(
+                    "Y-m-d H:i",
+                    $hold->requestExpirationDate
+                )
+                : null;
+            //set lastPickup Date if provided, format to j M Y
+            $lastPickup = isset($hold->holdShelfExpirationDate)
+                ? $this->dateConverter->convertToDisplayDate(
+                    "Y-m-d H:i",
+                    $hold->holdShelfExpirationDate
+                )
+                : null;
+            // setting some default availability / inTransit messages,
+            // in case they are not defined in Folio.ini
+            $availabilityArray = ['Open - Awaiting pickup'];
+            $inTransitArray = ['Open - In transit', 'Open - Awaiting delivery'];
             $holds[] = [
                 'type' => $hold->requestType,
-                'create' => date_format($requestDate, "j M Y"),
-                'expire' => isset($expireDate)
-                    ? date_format($expireDate, "j M Y") : "",
-                'id' => $this->getBibId(null, null, $hold->itemId),
+                'create' => $requestDate,
+                'expire' => $expireDate ?? "",
+                'id' => $this->getBibId(
+                    $hold->instanceId,
+                    $hold->holdingsRecordId,
+                    $hold->itemId
+                ),
                 'item_id' => $hold->itemId,
                 'reqnum' => $hold->id,
                 // Title moved from item to instance in Lotus release:
                 'title' => $hold->instance->title ?? $hold->item->title ?? '',
+                'available' => in_array(
+                    $hold->status,
+                    $this->config['Holds']['available'] ?? $availabilityArray
```

Comment:
This is great, but maybe a further improvement would be to move this to

```
/**
 * Default availability messages, in case they are not defined in Folio.ini
 *
 * @var string[]
 */
protected $defaultAvailabilityStatuses = ['Open - Awaiting pickup'];
```

at the top of the class, and then use `$this->defaultAvailabilityStatuses` in place of `$availabilityArray` here. (And of course do something similar for the other defaults as well). This not only makes the function code less cluttered by moving the variable definitions outside, but it also makes it possible to customize the values in a local subclass if there were ever a reason to do that.

It's a minor point, and fine as it is if you prefer to leave it... but just a suggestion!

Reply:
When it comes to creating code in a big project like this I am the most junior of junior developers, any guidance is gratefully appreciated. Despite the fact that these variables are unlikely to see much re-use (at least I cannot see that right now), I like the  thought of removing clutter from this function - so this will be in my next push.

I will just quickly run those changes on a test server with "live" data before hitting that push button.